MBBSEmu/HostProcess/MbbsHost.cs

```diff
@@ -977,10 +978,20 @@ private void DoNightlyCleanup()
         {
             _logger.Info("PERFORMING NIGHTLY CLEANUP");
 
+            // Notify Users of Nightly Cleanup
+            foreach (var c in _channelDictionary)
+                _channelDictionary[c.Value.Channel].SendToClient($"|RESET|\r\n|B||RED|Nightly Cleanup Running -- Please log back on shortly|RESET|\r\n".EncodeToANSIArray());
+            
             // removes all sessions
             RemoveSessions(session => true);
 
             CallModuleRoutine("mcurou", module => _logger.Info($"Calling nightly cleanup routine on module {module.ModuleIdentifier}"));
+            CallModuleRoutine("finrou", module => _logger.Info($"Calling finish-up (sys-shutdown) routine on module {module.ModuleIdentifier}"));
+            foreach (var m in _modules)
+            {
+                Run(m.Value.ModuleIdentifier, m.Value.EntryPoints["_INIT_"], ushort.MaxValue);
```

Comment:
We might need to add some additional checks on some things that happen at startup. One off the top of my head is text variable registration (`REGISTER_TEXTVAR`) and how MBBSEmu will happen if they're called again -- things like that we might need to make more resilient to do a Check/Add

Reply:
Testing with both variations (below) which fix crash with DIALCHAT. 

Trying a wide variety of modules and checking variables/module states before/after

**TryAdd:**
Module.TextVariables.TryAdd(Encoding.ASCII.GetString(textBytes), functionPointer);

**Update/Add:**
Module.TextVariables[Encoding.ASCII.GetString(textBytes)] = functionPointer;

Reply:
We'll want to verify anywhere a Dictionary is defined in the ExportedModule or the Base Class, that it is using a safe add method. I think there are only a couple, but just need to verify.